### ISSUE TITLE
Should `parsl-with-kubernetes` depend on `python-kubernetes`?

### ISSUE THREAD

**Issue by mfisher87** [conda-forge/parsl-feedstock#71]
I filled out this issue but the text was replaced when I submitted it. I don't really understand. Anyway, retyping:

If I install `parsl-with-kubernetes` and attempt to use it, I get:

```bash
Traceback (most recent call last):
  File "/home/mfisher/Projects/qgreenland-net/parsl-exploration/run.py", line 23, in <module>
    provider=KubernetesProvider(
             ^^^^^^^^^^^^^^^^^^^
  File "/home/mfisher/.miniconda3/envs/parsl-exploration/lib/python3.12/site-packages/parsl/providers/kubernetes/kube.py", line 106, in __init__
    raise OptionalModuleMissing(['kubernetes'],
parsl.errors.OptionalModuleMissing: The functionality requested requires optional modules ['kubernetes'] which could not be imported, because: Kubernetes provider requires kubernetes module and config.
```

Installing `python-kubernetes` resolves it. I think this should be added as a dependency to this recipe, but perhaps there is a reason it was omitted?

Would be happy to open a PR.

**Comment by xylar** [comment]
Ah, this is clearly an issue where the package on PyPI is called `kubernetes` but on conda-forge is called `python-kubernetes`.  Sorry about that!  Should be easy to fix.

**Comment by xylar** [comment]
This should be fixed in #72, and I also added better testing that would have caught this.

**Comment by mfisher87** [comment]
:zap:

Wonderful, thanks so much! I'm on a pair programming call right now and it was wonderful to see this resolved almost as quickly as we encountered it :star_struck:

I've been bitten by that same name-mismatch problem so many times as well. It's fun every time :laughing:
